Source/DafnyCore/Options/BoogieOptionBag.cs

```diff
@@ -58,6 +59,32 @@ public static class BoogieOptionBag {
     ArgumentHelpName = "seconds",
   };
 
+  public static readonly Option<int> VerificationErrorLimit =
+    new("--error-limit", () => 5, "Set the maximum number of errors to report (0 for unlimited).");
+
+  public static readonly Option<uint> SolverResourceLimit = new("--resource-limit",
+    @"Specify the maximum resource limit (rlimit) value to pass to Z3. A resource limit is a deterministic
+    alternative to a time limit. The output produced by `--log-format csv` includes the resource use of each proof
+    effort, which you can use to determine an appropriate limit for your program. Multiplied by 1000 before
+    sending to Z3.");
+  public static readonly Option<string> SolverPlugin = new("--solver-plugin",
+    @"Dafny uses Boogie as part of its verification process. This option allows customising that part using a Boogie plugin. More information about Boogie can be found at https://github.com/boogie-org/boogie. Information on how to construct Boogie plugins can be found by looking at the code in https://github.com/boogie-org/boogie/blob/v2.16.3/Source/Provers/SMTLib/ProverUtil.cs#L316");
+
+  public static readonly Option<string> SolverLog =
+    new("--solver-log", @"Specify a file to use to log the SMT-Lib text sent to the solver.") {
+      IsHidden = true
+    };
+
+  public static readonly Option<bool> IsolateAssertions = new("--isolate-assertions", @"Verify each assertion in isolation.");
+
+  public static readonly Option<FileInfo> SolverPath = new("--solver-path",
+    "Can be used to specify a custom SMT solver to use for verifying Dafny proofs.") {
+  };
+  public static readonly Option<IList<string>> SolverOption = new("--solver-option",
+    "Specify an option to control how the solver works.") {
```

Comment:
This line doesn't tell me how to use this. Could you mention that these arguments are passed directly to the solver CLI; that the default solver is Z3 with a link to its CLI documentation; and that Dafny is already configuring some options, so not all of options might be usable.

Reply:
I added a `--solver-option-help` flag that describes what all the options are, and expanded the help text for `--solver-option`.

Reply:
Thanks